src/WacomInterface.h

```diff
@@ -80,12 +81,45 @@ typedef struct {
 
 
 /**
- * General logging function. If priv is NULL, use a generic logging method,
- * otherwise priv is the device to log the message for.
+ * General logging function for a device.
  */
-__attribute__((__format__(__printf__ ,3, 4)))
+__attribute__((__format__(__printf__ , 3, 4)))
 void wcmLog(WacomDevicePtr priv, WacomLogType type, const char *format, ...);
 
+/**
+ * Same as the above but uses the sigsafe logging function if needed.
+ */
+__attribute__((__format__(__printf__ , 3, 4)))
+void wcmLogSafe(WacomDevicePtr priv, WacomLogType type, const char *format, ...);
```

Comment:
More for my general understanding than anything else, but how does one know when they're in a situation where the signal-safe versions are required?

Reply:
xserver 1.19 started using the input thread, before then we were using signal handlers for *all* input drivers (those calling `xf86AddEnabledDevice` which is all drivers but xf86-input-keyboard). And the sigsafe path is everything triggered from `dev->read_input`, so any event handling, etc. Only the properties code and the device setup/teardown code are outside of that path.

So, for the wacom driver - 90% of the code is in the signal handler path where we can't allocate, printf, etc.

xserver 1.19 and later we don't really need to worry about this unless someone disables the input thread during build, which I'm not sure anyone does.